internal/version/check_update.go

```diff
@@ -26,31 +45,102 @@ func CheckUpdate() {
 		return
 	}
 
-	githubClient := github.UnauthorizedClient()
-	release, _, err := githubClient.Repositories.GetLatestRelease(context.TODO(), repositoryOwner, repositoryName)
-	if err != nil {
-		logger.Debugf("Error: can't check latest release, %v", err)
+	v, ok := os.LookupEnv(checkUpdatedDisabledEnv)
+	if ok && strings.ToLower(v) != "false" {
+		logger.Debug("Disabled checking updates")
 		return
 	}
 
-	if release.TagName == nil || *release.TagName == "" {
-		logger.Debugf("Error: release tag is empty")
+	duration, err := time.ParseDuration(defaultCacheDuration)
+	if err != nil {
+		logger.Warnf("failed to parse cache duration: %s", err.Error())
 		return
 	}
+	latestVersion, expired, _ := checkCachedLatestVersion(duration)
+	var release *versionLatest
+	switch {
+	case !expired:
+		logger.Debugf("latest version (cached): %s", latestVersion)
+		release = latestVersion
+	default:
+		logger.Debugf("checking latest release in Github")
+		githubClient := github.UnauthorizedClient()
+		githubRelease, err := githubClient.LatestRelease(context.TODO(), repositoryOwner, repositoryName)
+		if err != nil {
+			logger.Debugf("Error: %v", err)
+			return
+		}
+		release = &versionLatest{
+			TagName:   *githubRelease.TagName,
+			HtmlURL:   *githubRelease.HTMLURL,
+			Timestamp: time.Now(),
+		}
+	}
 
 	currentVersion, err := semver.NewVersion(Tag[1:]) // strip "v" prefix
 	if err != nil {
 		logger.Debugf("Error: can't parse current version tag, %v", err)
 		return
 	}
 
-	releaseVersion, err := semver.NewVersion((*release.TagName)[1:]) // strip "v" prefix
+	releaseVersion, err := semver.NewVersion(release.TagName[1:]) // strip "v" prefix
 	if err != nil {
 		logger.Debugf("Error: can't parse current version tag, %v", err)
 		return
 	}
 
 	if currentVersion.LessThan(releaseVersion) {
-		logger.Infof("New version is available - %s. Download from: %s", *release.TagName, *release.HTMLURL)
+		logger.Infof("New version is available - %s. Download from: %s", release.TagName, release.HtmlURL)
+	}
+
+	if err := writeLatestReleaseToCache(release); err != nil {
+		logger.Debugf("failed to write latest versoin to cache file: %v", err)
+	}
+}
+
+func writeLatestReleaseToCache(release *versionLatest) error {
+	elasticPackagePath, err := locations.NewLocationManager()
+	if err != nil {
+		return fmt.Errorf("failed locating the configuration directory: %w", err)
+	}
+
+	latestVersionPath := filepath.Join(elasticPackagePath.RootDir(), latestVersionFile)
+
+	contents, err := json.Marshal(release)
+	if err != nil {
+		return fmt.Errorf("failed to encode file %s: %w", latestVersionPath, err)
+	}
+	err = os.WriteFile(latestVersionPath, contents, 0644)
+	if err != nil {
+		return fmt.Errorf("writing file failed (path: %s): %w", latestVersionPath, err)
+	}
+
+	return nil
+}
+
+func checkCachedLatestVersion(expiration time.Duration) (*versionLatest, bool, error) {
```

Comment:
Nit. Split this method in one to load the latest version, and another one to check it?
```suggestion
func loadCachedLatestVersion() (*versionLatest, error) {
    ...
}

func checkLatestVersion(latest *versionLatest, expiration time.Duration) bool {
    return latest.Timestamp.Before(expiration)
}
```